impl AsRef for *

It'd be easier to throw this at things if it had `AsRef<Path>` everywhere or something.

```
error[E0277]: the trait bound `ChildPath: AsRef<Path>` is not satisfied
   --> src/git/tests.rs:32:45
    |
32  |         let git = Git::at_path(Initializer, &repo);
    |                                             ^^^^^ the trait `AsRef<Path>` is not implemented for `ChildPath`
    | 
   ::: src/git.rs:211:23
    |
211 |     pub fn at_path<T: AsRef<Path>>(_: Initializer, path: T) -> Option<Git> {
    |                       ----------- required by this bound in `git::Git::at_path`
    |
    = note: required because of the requirements on the impl of `AsRef<Path>` for `&ChildPath`

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 * Explicitly stated that `assert` may `panic`
 
+#### Features
+
+* Built-in fixtures now interoperate as `&Path`
+
 ## [1.0.1] - 2021-02-01
 
 ## 1.0.0 - 2020-03-26

--- a/src/fixture/child.rs
+++ b/src/fixture/child.rs
@@ -89,3 +89,17 @@ impl ChildPath {
         &self.path
     }
 }
+
+impl AsRef<path::Path> for ChildPath {
+    fn as_ref(&self) -> &path::Path {
+        &self.path
+    }
+}
+
+impl std::ops::Deref for ChildPath {
+    type Target = path::Path;
+    #[inline]
+    fn deref(&self) -> &path::Path {
+        &self.path
+    }
+}

--- a/src/fixture/dir.rs
+++ b/src/fixture/dir.rs
@@ -191,3 +191,17 @@ impl TempDir {
         Ok(())
     }
 }
+
+impl AsRef<path::Path> for TempDir {
+    fn as_ref(&self) -> &path::Path {
+        self.path()
+    }
+}
+
+impl std::ops::Deref for TempDir {
+    type Target = path::Path;
+    #[inline]
+    fn deref(&self) -> &path::Path {
+        self.path()
+    }
+}

--- a/src/fixture/file.rs
+++ b/src/fixture/file.rs
@@ -192,3 +192,17 @@ impl NamedTempFile {
         Ok(())
     }
 }
+
+impl AsRef<path::Path> for NamedTempFile {
+    fn as_ref(&self) -> &path::Path {
+        self.path()
+    }
+}
+
+impl std::ops::Deref for NamedTempFile {
+    type Target = path::Path;
+    #[inline]
+    fn deref(&self) -> &path::Path {
+        self.path()
+    }
+}

